nats.go

```diff
@@ -577,6 +577,9 @@ type Subscription struct {
 	// Type of Subscription
 	typ SubscriptionType
 
+	// Whether subject in messages and Subject in subscription may differ
+	subjectMayDiffer bool
```

Comment:
would prefer short variable style for the optimization along with the comment, maybe:
```suggestion
	smd bool
```

Reply:
Why not